js/adapt-assessmentQuestionBank.js

```diff
@@ -26,8 +27,13 @@ class QuestionBank {
   }
 
   getRandomQuestionBlocks() {
+    if (this._count > this.unusedQuestionBlocks.length) {
+      logging.warn(`Bank ID ${this._bankId}: Attempting to display ${this._count} question(s), but only ${this.unusedQuestionBlocks.length} available`);
+    }
+
     const questionBlocks = [];
     let i = 0;
+
     while (i++ < this._count) {
```

Comment:
At the moment if this._count is higher than the number of questions I'm getting the warning but still getting a duplicate question. Thinking something along the lines of below will stop that behavior. 

```
let count = this._count;
if (this._count > this.unusedQuestionBlocks.length) {
  count = this.unusedQuestionBlocks.length;
  logging.warn(`Bank ID ${this._bankId}: Attempting to display ${this._count} question(s), but only ${this.unusedQuestionBlocks.length} available`);
}

const questionBlocks = [];
let i = 0;

while (i++ < count) {
  const nextBlock = this.unusedQuestionBlocks.shift();
  questionBlocks.push(nextBlock);
}
return questionBlocks;
```

Reply:
Excellent! Thanks